operations/service_deploy_smoke_utils.go

```diff
@@ -93,14 +95,57 @@ func getLatestGithubCommit() (string, error) {
 	return "", errors.New("could not find latest commit in response")
 }
 
-func checkTaskByCommit(username, key string) error {
+func checkContainerTask(username, key string) error {
+	client := utility.GetHTTPClient()
+	defer utility.PutHTTPClient(client)
+
+	var task apimodels.APITask
+OUTER:
+	for i := 0; i <= 30; i++ {
+		// check task
+		if i == 30 {
+			return errors.Errorf("task status is %s (expected %s)", task.Status, evergreen.TaskSucceeded)
+		}
+		time.Sleep(10 * time.Second)
+		grip.Infof("checking for task '%s' (%d/30)", containerTaskId, i+1)
+
+		var err error
+		task, err = checkTask(client, username, key, containerTaskId)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+
+		if task.Status == evergreen.TaskFailed {
+			return errors.Errorf("task status is %s (expected %s)", task.Status, evergreen.TaskSucceeded)
+		}
+		if task.Status != evergreen.TaskSucceeded {
+			grip.Infof("found task is status %s", task.Status)
+			task = apimodels.APITask{
```

Comment:
This isn't wrong, but it's written kind of strangely because the only reason the loop declares `task` outside the for block is to interpolate the status on L107. Can we simplify this down a bit to just set a string for the task status after `checkTask` rather than recycling the entire task between iterations?